src/server/api/v1/utils.py

```diff
@@ -0,0 +1,102 @@
+import datetime
+
+import jwt
+from werkzeug.security import generate_password_hash, check_password_hash
+
+from api.v1.db import db_helper
+from api.v1.db.db_models import UserModel
+from api.v1.config import SECRET_KEY
+
+
+def encode_auth_token(user_id):
+    try:
+        payload = {
+            'exp': datetime.datetime.utcnow() + datetime.timedelta(minutes=10),
+            'iat': datetime.datetime.utcnow(),
+            'sub': user_id
+        }
+        return jwt.encode(
+            payload,
+            SECRET_KEY,
+            algorithm='HS256')
+    except Exception as e:
+        return e
+
+
+def decode_auth_token(auth_token):
+    try:
+        payload = jwt.decode(auth_token, SECRET_KEY)
+        return payload['sub']
+    except jwt.ExpiredSignatureError:
+        return "Signature expired. Please sign-in again"
+    except jwt.InvalidTokenError:
+        return "Invalid token. Please sign-in again"
+
+
+def default(req):
+    req.serv_reply(200, {'message': 'Welcome to Warship Battle! Visit \'/register\' to get registered!'})
+
+
+def path_not_found(req):
+    req.serv_reply(404, {'message': 'Requested path is not found, please check the endpoint and try again'})
+
+
+def login(req, data):
+    user = db_helper.db_connection.query(UserModel).filter_by(nickname=data['nickname']).first()
+    if not (db_helper.UserDBHelper.user_exists(data) and check_password_hash(user.passwd, data['password'])):
+        req.serv_reply(401, {'message': 'Unauthorized. Username or password is invalid'})
+    else:
+        token = encode_auth_token(user.id)
+
+        # data to be updated in the DB {<field_name>: <value>}
+        data = {'token': token}
+        db_helper.UserDBHelper.update(user.id, data)
+        db_helper.db_connection.commit()
+        req.serv_reply(200, {'token': token.decode('utf-8')})
+
+
+def register(req, data):
+    if not db_helper.UserDBHelper.user_exists(data):
+        user_obj = db_helper.UserDBHelper(db_helper.db_connection)
+        passwd = generate_password_hash(data['password'])
+        user_obj.create(data['nickname'], passwd)
+        req.serv_reply(201, {'message': 'User successfully created'})
+    else:
+        req.serv_reply(409, {'message': 'Username is already taken'})
+
+
+def users(req):
+    users = db_helper.db_connection.query(UserModel).all()
+    response = {'users': {}}
+    for i in range(len(users)):
+        data = {'id': users[i].id, 'name': users[i].nickname}
+        response['users'][i] = data
+    req.serv_reply(200, response)
```

Comment:
Why didn`t use list, for example:
response = {'users': []}
for user in users:
        data = {'id': user.id, 'name': user.nickname}
        response['users'].append(data)

Reply:
Done